src/rpc_client/eth_rpc/mod.rs

```diff
@@ -0,0 +1,851 @@
+
+    /// The max number of RPC call retries we expect to see (plus one).
+    const MAX_EXPECTED_RETRIES: usize = 20;
+
+    #[derive(Default)]
+    struct RetryHistogram {
+        /// The histogram of HTTP call retry counts.
+        /// The last bucket corresponds to the "infinite" value that exceeds the maximum number we
+        /// expect to see in practice.
+        retry_buckets: [u64; MAX_EXPECTED_RETRIES + 1],
+        retry_count: u64,
+    }
+
+    impl RetryHistogram {
+        fn observe_retry_count(&mut self, count: usize) {
+            self.retry_buckets[count.min(MAX_EXPECTED_RETRIES)] += 1;
+            self.retry_count += count as u64;
+        }
+
+        /// Returns a iterator over the histrogram buckets in the format that ic-metrics-encoder
+        /// expects.
+        fn iter(&self) -> impl Iterator<Item = (f64, f64)> + '_ {
+            (0..MAX_EXPECTED_RETRIES)
+                .zip(self.retry_buckets[0..MAX_EXPECTED_RETRIES].iter().cloned())
+                .map(|(k, v)| (k as f64, v as f64))
+                .chain(std::iter::once((
+                    f64::INFINITY,
+                    self.retry_buckets[MAX_EXPECTED_RETRIES] as f64,
+                )))
+        }
+    }
+
+    #[derive(Default)]
+    pub struct HttpMetrics {
+        /// Retry counts histograms indexed by the ETH RCP method name.
+        retry_histogram_per_method: BTreeMap<String, RetryHistogram>,
+    }
+
+    impl HttpMetrics {
+        pub fn observe_retry_count(&mut self, method: String, count: usize) {
+            self.retry_histogram_per_method
+                .entry(method)
+                .or_default()
+                .observe_retry_count(count);
+        }
+
+        #[cfg(test)]
+        pub fn count_retries_in_bucket(&self, method: &str, count: usize) -> u64 {
+            match self.retry_histogram_per_method.get(method) {
+                Some(histogram) => histogram.retry_buckets[count.min(MAX_EXPECTED_RETRIES)],
+                None => 0,
+            }
+        }
+
+        pub fn encode<W: std::io::Write>(
+            &self,
+            encoder: &mut MetricsEncoder<W>,
+        ) -> std::io::Result<()> {
+            if self.retry_histogram_per_method.is_empty() {
+                return Ok(());
+            }
+
+            let mut histogram_vec = encoder.histogram_vec(
+                "cketh_eth_rpc_call_retry_count",
+                "The number of ETH RPC call retries by method.",
+            )?;
+
+            for (method, histogram) in &self.retry_histogram_per_method {
+                histogram_vec = histogram_vec.histogram(
+                    &[("method", method.as_str())],
+                    histogram.iter(),
+                    histogram.retry_count as f64,
+                )?;
+            }
+
+            Ok(())
+        }
+    }
+
+    //TODO XC-243: use existing METRICS declared in memory.rs
+    thread_local! {
+        static METRICS: RefCell<HttpMetrics> = RefCell::default();
+    }
```

Comment:
@rvanasa : Do we even need those metrics? The current version of the [EVM RPC](https://7hfb6-caaaa-aaaar-qadga-cai.raw.icp0.io/metrics) canister doesn't expose them. This is meaningful for ckETH, to determine if the default response size is appropriate, but in case of the EVM RPC canister, where user could override the default, I don't know if that's too meaningful. WDYT?

Reply:
We can remove the ckETH metrics (imo).

Reply:
Ok, then I will remove those in a separate PR.